src/Sylius/Component/Core/Promotion/Action/UnitPercentageDiscountPromotionActionCommand.php

```diff
@@ -75,4 +82,31 @@ private function setUnitsAdjustments(
             $this->addAdjustmentToUnit($unit, $promotionAmount, $promotion);
         }
     }
+
+    /**
+     * @param array<string, mixed> $configuration
+     * @return array<string, mixed>
```

Comment:
Looks like `filteredItems` will have numeric indices for keys, so for example
```suggestion
     * @return array<array-key, mixed>
```